manifests/init.pp

```diff
@@ -345,6 +351,9 @@
   Integer[1, 65535] $ssl_port                                                                      = 5671,
   Optional[String] $ssl_interface                                                                  = undef,
   Integer[1, 65535] $ssl_management_port                                                           = 15671,
+  Optional[Stdlib::Absolutepath] $ssl_management_cacert                                            = $ssl_cacert,
+  Optional[Stdlib::Absolutepath] $ssl_management_cert                                              = $ssl_cert,
+  Optional[Stdlib::Absolutepath] $ssl_management_key                                               = $ssl_key,
```

Comment:
@bastelfreak: can a param referring to another param that defaults to undef like this be optional?

Reply:
after the referrenced variables are undef by default, the new variables also has to be optional.  

puppet documentation "Optional":

> The Optional data type wraps one other data type, and results in a data type that matches anything that type would match plus undef.

https://puppet.com/docs/puppet/5.3/lang_data_abstract.html#optional

Reply:
Yeah, I just wasn’t sure if it would chain properly. Should be fine.